sources/bing_webmaster/__init__.py

```diff
@@ -0,0 +1,121 @@
+"""
+A source loading history of organic search traffic from Bing Webmaster API
+See documentation: https://learn.microsoft.com/en-us/dotnet/api/microsoft.bing.webmaster.api.interfaces.iwebmasterapi?view=bing-webmaster-dotnet
+"""
+
+from typing import Any, Dict, Iterator, Iterable, List, Sequence
+
+import dlt
+from dlt.sources import DltResource
+from dlt.common.typing import DictStrStr, DictStrAny
+import time
+
+from .helpers import get_stats_with_retry, parse_response
+
+
+@dlt.source(name="bing_webmaster")
+def source(
+    site_urls: List[str] = None, site_url_pages: Iterable[DictStrStr] = None
+) -> Sequence[DltResource]:
+    """
+    A dlt source for the Bing Webmaster api.
+    It groups resources for the APIs which return organic search traffic statistics
+    Args:
+        site_urls: List[str]: A list of site_urls, e.g, ["dlthub.com", "dlthub.de"]. Use this if you need the weekly traffic per site_url and page
+        site_url_pages: Iterable[DictStrStr]: A list of pairs of site_url and page. Use this if you need the weekly traffic per site_url, page, and query
+    Returns:
+        Sequence[DltResource]: A sequence of resources that can be selected from including page_stats and page_query_stats.
+    """
+    return (
```

Comment:
Thank you! Did you have the following in mind?
```python
yield (
        page_stats(site_urls),
        page_query_stats(site_url_pages),
    )
```

I'm not sure because both the boilerplate code as well as all verified sources I checked return a tuple or list of resources, I couldn't find a source yet which yields resources.

Reply:
it's required that data from resources must be yielded, but it's not necessary to yield resources

Reply:
you can yield resources if you need to return them with pipe for example, look at pipedrive: https://github.com/dlt-hub/verified-sources/blob/master/sources/pipedrive/__init__.py

Reply:
Good! But I think that the data is yielded from the resources. Or can a yield not call a helper function which returns? Please advise.